src/twisted/internet/base.py

```diff
@@ -335,7 +335,7 @@ def getHostByName(
             timeoutDelay = sum(timeout)
         else:
             timeoutDelay = 60
-        userDeferred = defer.Deferred()  # type: Deferred[str]
+        userDeferred: Deferred[str] = defer.Deferred()
```

Comment:
I think this form is preferred:

```suggestion
        userDeferred = defer.Deferred[str]()
```

Reply:
Even better, get rid of the `defer.` prefix